src/applications/vaos/services/appointment/index.js

```diff
@@ -245,37 +245,43 @@ export async function fetchRequestById({ id, useV2 }) {
  * @param {'cc'|'va'} type Type of appointment that is being fetched
  * @returns {Appointment} A transformed appointment with the given id
  */
-export async function fetchBookedAppointment(id, type) {
+export async function fetchBookedAppointment({ id, type, useV2 = false }) {
   try {
     let appointment;
-    if (type === 'va') {
-      appointment = await getConfirmedAppointment(id, type);
-    } else if (type === 'cc') {
-      // We don't have a fetch by id service for cc, so hopefully
-      // the appointment is 13 months in either direction
-      const { data } = await getConfirmedAppointments(
-        type,
-        moment()
-          .add(-395, 'days')
-          .startOf('day')
-          .toISOString(),
-        moment()
-          .add(395, 'days')
-          .startOf('day')
-          .toISOString(),
-      );
-      appointment = data.find(appt => appt.id === id);
-
+    if (!useV2) {
+      if (type === 'va') {
+        appointment = await getConfirmedAppointment(id, type);
+      } else if (type === 'cc') {
+        // We don't have a fetch by id service for cc, so hopefully
+        // the appointment is 13 months in either direction
+        const { data } = await getConfirmedAppointments(
+          type,
+          moment()
+            .add(-395, 'days')
+            .startOf('day')
+            .toISOString(),
+          moment()
+            .add(395, 'days')
+            .startOf('day')
+            .toISOString(),
+        );
+        appointment = data.find(appt => appt.id === id);
+
+        if (!appointment) {
+          appointment = await getConfirmedAppointment(id, 'va');
+        }
+      }
       if (!appointment) {
-        appointment = await getConfirmedAppointment(id, 'va');
+        throw new Error(`Couldn't find ${type} appointment`);
       }
+      return transformConfirmedAppointment(appointment);
+    } else {
+      appointment = await getAppointment(id);
+      if (!appointment) {
```

Comment:
You can leave out this if statement and throw, if it can't be found it'll throw an error automatically